### PR TITLE
Add a possibility to switch back to building images by secret

### PR DESCRIPTION
**Pull request by potiuk**
You can now define secret in your own fork:

AIRFLOW_GITHUB_REGISTRY_WAIT_FOR_IMAGE

If you set it to "false", it skips building images in separate
workflow_run - images will be built in the jobs run in the
CI Build run and they won't be pushed to the registry.

Note - you can't have secrets starting with GITHUB_, that's why
the AIRFLOW_* prefix

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -56,9 +56,10 @@ env:
   #
   #  GITHUB_REGISTRY_WAIT_FOR_IMAGE: "false"
   #
+  # You can also switch back to building images locally and disabling the "Build Images" workflow
+  # by defining AIRFLOW_GITHUB_REGISTRY_WAIT_FOR_IMAGE secret with value set to "false"
 
-  GITHUB_REGISTRY_WAIT_FOR_IMAGE: "true"
-
+  GITHUB_REGISTRY_WAIT_FOR_IMAGE: ${{ secrets.AIRFLOW_GITHUB_REGISTRY_WAIT_FOR_IMAGE != 'false' }}
```
Comment (feluelle) [inline .github/workflows/ci.yml]:
```suggestion
  GITHUB_REGISTRY_WAIT_FOR_IMAGE: ${{ secrets.AIRFLOW_GITHUB_REGISTRY_WAIT_FOR_IMAGE }}
```

File: .github/workflows/build-images-workflow-run.yml [hunk 2]
```diff
@@ -39,6 +39,7 @@ env:
   GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
   GITHUB_REGISTRY_PULL_IMAGE_TAG: "latest"
   GITHUB_REGISTRY_WAIT_FOR_IMAGE: "false"
+  BUILD_IMAGES: ${{ secrets.AIRFLOW_GITHUB_REGISTRY_WAIT_FOR_IMAGE != 'false' }}
```
Comment (feluelle) [inline .github/workflows/build-images-workflow-run.yml]:
```suggestion
  BUILD_IMAGES: ${{ secrets.AIRFLOW_GITHUB_REGISTRY_WAIT_FOR_IMAGE }}
```

**Comment by potiuk** [comment]
Indeed @feluelle -> it should work this way :)

**Comment by potiuk** [comment]
Well. not really @feluelle :(. The problem is when you have not defined secret - then I wanted to behave it as if it was "true" - now it 'is "empty" :(

**Review by feluelle — APPROVED** [review]
I see. LGTM then.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (potiuk) [reply .github/workflows/ci.yml]: As explained below - I only want the secret to trigger the "BUILD_IMAGE=false" if it is defined and it has "false" value. So the condition is actually OK :)

Comment (potiuk) [reply .github/workflows/ci.yml]: I pushed back the original change

File: .github/workflows/build-images-workflow-run.yml (hunk 2 above)
Comment (potiuk) [reply .github/workflows/build-images-workflow-run.yml]: Same here

**Comment by potiuk** [comment]
I also explain that in #10380 where I keep the docs updated.

**Comment by mik-laj** [comment]
I would like to do a review today.

**Comment by potiuk** [comment]
@mik-laj of this PR or the Doc one ?

**Comment by mik-laj** [comment]
Both. :-D

**Comment by mik-laj** [comment]
Should we add some docs about it?

**Comment by mik-laj** [comment]
@potiuk Can you create a new PR? I accidentally broke this one when I tested it.

**Comment by potiuk** [comment]
> Should we add some docs about it?

See the comment I made above:

> I also explain that in #10380 where I keep the docs updated.

**Comment by potiuk** [comment]
PR #10509

**Pull request closed without merging**